python/tests/integration/server_manager.py

```diff
@@ -108,7 +108,8 @@ def client_connection(self):
             )
             yield conn
         finally:
-            conn.close()
+            if 'conn' in locals():
```

Comment:
No. Initialize it `None` outside the `try` block and check if it's `not None` here.